redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseApp.kt

```diff
@@ -183,22 +186,41 @@ public class TreehouseApp<A : AppService> private constructor(
     eventPublisher.appCanceled()
   }
 
-  private class ZiplineCodeHost<A : AppService>(
-    private val appScope: CoroutineScope,
-    private val dispatchers: TreehouseDispatchers,
-    private val eventPublisher: EventPublisher,
-    private val frameClock: FrameClock,
-    override val stateStore: StateStore,
-  ) : CodeHost<A> {
-    override var session: ZiplineCodeSession<A>? = null
+  private class TreehouseDispatchersWithExceptionHandler(
+    val delegate: TreehouseDispatchers,
+    exceptionHandler: CoroutineExceptionHandler,
+  ) : TreehouseDispatchers by delegate {
+    override val zipline = delegate.zipline + exceptionHandler
+  }
 
+  private inner class ZiplineCodeHost<A : AppService> : CodeHost<A> {
```

Comment:
Was their a requirement to make this an `inner class`?

(My over dislike of `inner class` may have motivated this comment 😅)

Reply:
It reduces the amount of boilerplate to share state with the enclosing class.